apps/settings/js/panels/call/call_forwarding.js

```diff
@@ -0,0 +1,202 @@
+  'use strict';
+
+  var Module = require('modules/base/module');
+  var EventEmitter = require('modules/base/event_emitter');
+  var CallUtils = require('panels/call/call_utils');
+  var CallConstant = require('panels/call/call_constant');
+
+  /**
+   * @class CallForwarding
+   * @returns {CallForwarding}
+   */
+  var CallForwarding = Module.create(function() {
+    this.super(EventEmitter).call(this, [
+      'set-callForwarding-invalid',
+      'set-callForwarding-updating',
+      'set-callForwarding-success',
+      'set-callForwarding-error',
+      'update-callForwarding-subpanels',
+      'enable-menuItems',
+      'update-all-desc-info-with-l10n-id',
+      'update-all-desc-info-with-rules'
+    ]);
+
+    this._conn = null;
+
+    /** Stores current states (enabler or not) of the call forwaring reason */
+    this._cfReasonStates = [0, 0, 0, 0];
+  }).extend(EventEmitter);
+
+  CallForwarding.prototype.setCallForwardingValues = function(options) {
+    var key = options.key;
+    var number = options.number;
+    var enabled = !!options.enabled;
+
+    // While storing the settings into the database we avoid observing
+    // changes on those ones and enabling/disabling call forwarding.
+    //
+    // TODO we have to add this back later
+    if (!key) {
+      return;
+    } else if (this._cfReasonStates[
+        CallConstant.CALL_FORWARD_REASON_MAPPING[key]] === enabled) {
+      // Bails out in case the reason is already enabled/disabled.
+      return;
+    } else {
+      var mozMobileCFInfo = {};
+      mozMobileCFInfo.action = enabled ?
+        CallConstant.CALL_FORWARD_ACTION.REGISTRATION :
+        CallConstant.CALL_FORWARD_ACTION.DISABLE;
+      mozMobileCFInfo.reason =
+        CallConstant.CALL_FORWARD_REASON_MAPPING[key];
+      mozMobileCFInfo.serviceClass = this._conn.ICC_SERVICE_CLASS_VOICE;
+
+      if (!CallUtils.isPhoneNumberValid(number)) {
+        this._emitEvent('set-callForwarding-invalid');
+      } else {
+        mozMobileCFInfo.number = number;
+        mozMobileCFInfo.timeSeconds =
+          mozMobileCFInfo.reason !=
+            CallConstant.CALL_FORWARD_REASON.NO_REPLY ? 0 : 20;
+
+        var req = this._conn.setCallForwardingOption(
+          mozMobileCFInfo);
+
+        this._emitEvent('set-callForwarding-updating');
+
+        req.onsuccess = () => {
+          this._emitEvent('set-callForwarding-success', {
+            key: key,
+            action: mozMobileCFInfo.action
+          });
+        };
+
+        req.onerror = () => {
+          this._emitEvent('set-callForwarding-error');
+        };
+      }
+    }
+  };
+
+  /**
+   * We can use this function to get specific call forwarding option.
+   *
+   * @memberOf CallForwarding
+   * @param {Number} cardIndex
+   * @param {String} optionName
+   * @return {Promise}
+   */
+  CallForwarding.prototype.getOption = function(cardIndex, optionName) {
+    this._conn = window.navigator.mozMobileConnections[cardIndex];
```

Comment:
Setting `this._conn` here seems weird to me. That means the user of this module should call `getOption` firs before calling to other functions that use mobile connection.